src/WebJobs.Script/Description/DotNet/DependencyHelper.cs

```diff
@@ -5,32 +5,40 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Reflection;
 using System.Runtime.InteropServices;
+using System.Text.Json;
+using Microsoft.Azure.WebJobs.Script.Description.DotNet;
 using Microsoft.Azure.WebJobs.Script.ExtensionRequirements;
 using Microsoft.Extensions.DependencyModel;
-using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Azure.WebJobs.Script.Description
 {
     public static class DependencyHelper
     {
         private const string AssemblyNamePrefix = "assembly:";
-        private static readonly Lazy<Dictionary<string, string[]>> _ridGraph = new Lazy<Dictionary<string, string[]>>(BuildRuntimesGraph);
+        private static readonly Assembly ThisAssembly = typeof(DependencyHelper).Assembly;
+        private static readonly string ThisAssemblyName = ThisAssembly.GetName().Name;
+        private static readonly Lazy<Dictionary<string, string[]>> RidGraph = new Lazy<Dictionary<string, string[]>>(BuildRuntimesGraph);
+
         private static string _runtimeIdentifier;
 
         private static Dictionary<string, string[]> BuildRuntimesGraph()
         {
-            var ridGraph = new Dictionary<string, string[]>();
-            string runtimesJson = GetRuntimesGraphJson();
-            var runtimes = (JObject)JObject.Parse(runtimesJson)["runtimes"];
+            using var stream = GetEmbeddedResourceStream("runtimes.json");
+
+            var runtimeGraph = JsonSerializer.Deserialize(stream, RuntimeGraphJsonContext.Default.RuntimeGraph);
 
-            foreach (var runtime in runtimes)
+            if (runtimeGraph is not { Runtimes.Count: > 0 })
             {
-                string[] imports = ((JObject)runtime.Value)["#import"]
-                    ?.Values<string>()
-                    .ToArray();
+                throw new InvalidOperationException("Failed to deserialize runtimes graph JSON or runtimes section is empty.");
```

Comment:
I'm not sure about the behavior, but now we're throwing an exception. Will this affect existing apps in the bad state? There's another exception in `ScriptRuntimeAssembly` method

Reply:
This file is embedded with the host, and its contents are not being modified in this PR. Therefore, I expect this method to continue functioning as before. If the file is missing or corrupted, it's better to fail fast during startup since this is a critical file required for proper operation.